src/platform/silabs/efr32/BLEChannel.h

```diff
@@ -0,0 +1,188 @@
+#pragma once
+
+#include "sl_bt_api.h"
+#include "sl_bt_rtos_adaptation.h"
+#include "sl_bt_stack_config.h"
+#include "sl_bt_stack_init.h"
+#include <lib/core/CHIPError.h>
+#include <lib/core/Optional.h>
+#include <lib/support/BitFlags.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace DeviceLayer {
+namespace Internal {
+
+struct BLEConState
+{
+    uint16_t mtu : 10;
+    uint16_t allocated : 1;
+    uint16_t subscribed : 1;
+    uint16_t unused : 4;
+    uint8_t connectionHandle;
+    uint8_t bondingHandle;
+};
+
+struct AdvConfigStruct
+{
+    ByteSpan advData;
+    ByteSpan responseData;
+    uint32_t intervalMin;
+    uint32_t intervalMax;
+    uint16_t duration;
+    uint8_t maxEvents;
+};
+
+class BLEChannel
+{
+public:
+    BLEChannel()
+    {
+        // Initialize the connection state
+        memset(&mConnectionState, 0, sizeof(mConnectionState));
+        mFlags.ClearAll();
+        mAdvIntervalMin = 0;
+        mAdvIntervalMax = 0;
+        mAdvDuration    = 0;
+        mAdvMaxEvents   = 0;
```

Comment:
Is this necessary in the constructor? default values already configure this so this seems like unnecessary steps.